op-node/node/node.go

```diff
@@ -159,27 +159,70 @@ func (n *OpNode) initRuntimeConfig(ctx context.Context, cfg *Config) error {
 	// attempt to load runtime config, repeat N times
 	n.runCfg = NewRuntimeConfig(n.log, n.l1Source, &cfg.Rollup)
 
-	for i := 0; i < 5; i++ {
+	confDepth := cfg.Driver.VerifierConfDepth
+	reload := func(ctx context.Context) (eth.L1BlockRef, error) {
 		fetchCtx, fetchCancel := context.WithTimeout(ctx, time.Second*10)
 		l1Head, err := n.l1Source.L1BlockRefByLabel(fetchCtx, eth.Unsafe)
 		fetchCancel()
 		if err != nil {
 			n.log.Error("failed to fetch L1 head for runtime config initialization", "err", err)
-			continue
+			return eth.L1BlockRef{}, err
+		}
+
+		// Apply confirmation-distance
+		blNum := l1Head.Number
+		if blNum >= confDepth {
+			blNum -= confDepth
+		}
+		fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
+		confirmed, err := n.l1Source.L1BlockRefByNumber(fetchCtx, blNum)
+		fetchCancel()
+		if err != nil {
+			n.log.Error("failed to fetch confirmed L1 block for runtime config loading", "err", err, "number", blNum)
+			return eth.L1BlockRef{}, err
 		}
 
 		fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
-		err = n.runCfg.Load(fetchCtx, l1Head)
+		err = n.runCfg.Load(fetchCtx, confirmed)
 		fetchCancel()
 		if err != nil {
 			n.log.Error("failed to fetch runtime config data", "err", err)
-			continue
+			return l1Head, err
 		}
+		return l1Head, nil
+	}
 
-		return nil
+	// initialize the runtime config before unblocking
+	if _, err := retry.Do(ctx, 5, retry.Fixed(time.Second*10), func() (eth.L1BlockRef, error) {
+		return reload(ctx)
+	}); err != nil {
+		return fmt.Errorf("failed to load runtime configuration repeatedly, last error: %w", err)
 	}
 
-	return errors.New("failed to load runtime configuration repeatedly")
+	// start a background loop, to keep reloading it every 10 minutes
```

Comment:
nit: It's not a fixed 10 minute duration now.
```suggestion
	// start a background loop, to keep reloading it at the configured reload interval
```